terraform/README.md

```diff
@@ -1,18 +1,32 @@
-# Terraform modules for Digital Marketplace
+# Terraform for the Digital Marketplace
 
-## Root modules
+## Projects and modules
 
- * **aws-dm**: digitalmarketplace AWS account
- * **aws-dm-dev**: digitalmarketplace-development AWS account
- * **aws-dm-prod**: digitalmarketplace-production AWS account
+The following projects each define and control the state of a section of our infrastructure.
+
+ * **accounts/development**: The digitalmarketplace-development AWS account
+ * **accounts/main**: The digitalmarketplace AWS account
+ * **accounts/production**: The digitalmarketplace-production AWS account
+ * **environments/preview**
+ * **environments/staging**
+ * **environments/production**
+
+Account projects apply to the AWS account as a whole, so should only contain resources that are created once per
+account (eg root Route53 hosted zones, IAM users).
+
+Environment projects apply to individual environments and should contain resources that should be created for each
+environment (eg application logs and S3 buckets).
+
+All projects are then able to import and use the functionality they require from our folder of shared modules.
 
 ## Requirements
```

Comment:
On the top level README I think it's worth making it clear that all these bits need to be installed separately (i.e. not with pip), not just Terraform.

Reply:
By installed do you mean with `make`? I'm not sure I realised that! - isn't it enough to run `make requirements` at the top level? The other makefiles are then specific to actually deploying things into AWS, rather than building local environments / build artefacts?

(I must admit I find it a bit confusing having makefiles at multiple levels, not sure I want to suggest a remedy for that though :-) )

Reply:
I've added a commit that I hope makes this a bit clearer.